Scripts/Druid: Implement Flourish

Nyr97 opened the pull request:
**Changes proposed:**

- Implement Flourish talent.

**Issues addressed:**

None.

**Tests performed:**

Tested and it works.

**Known issues and TODO list:**

None.

Nyr97 committed:
Scripts/Druid: Implement Flourish

Nyr97 committed:
update

Seyden commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -559,6 +559,38 @@ class spell_dru_ferocious_bite : public SpellScript
+            if (aurEff->GetCasterGUID() != caster->GetGUID() || aurEff->IsAffectingSpell(GetSpellInfo()))
+                continue;
+
+            if (Aura* aura = aurEff->GetBase())
```
How is it possible that the aura will be null here?

Nyr97 committed:
update

Nyr97 committed:
update

Nyr97 committed:
update 2

Nyr97 commented:
I updated the old script so that we can filter out targets that were not supposed to be affected anyway.

Nyr97 committed:
update 3

Nyr97 committed:
update 4

Nyr97 commented:
@Shauren old, but valid for review

Nyr97 committed:
update 5

Nyr97 committed:
last update

std::unordered_map::find is safe to use on an empty map, so if the map is empty, find will immediately return end(), so we're gucci

Nyr97 committed:
final update 2

Nyr97 committed:
Scripts/Druid: Update script

Nyr97 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -883,6 +883,72 @@ class spell_dru_ferocious_bite : public SpellScript
+                continue;
+
+            RegisterAffectedAurasForTarget(unitTarget->GetGUID(), aura);
+            hasValidAura = true;
```
not sure if it'd be any better in terms of performance to check for the value to be false to be set to true, should be the same overhead but whatever
